sway/desktop/launcher.c

```diff
@@ -151,12 +187,13 @@ void root_record_workspace_pid(pid_t pid) {
 	}
 
 	struct pid_workspace *pw = calloc(1, sizeof(struct pid_workspace));
-	pw->workspace = strdup(ws->name);
```

Comment:
Hm. I think we record the workspace here in case the user goes to workspace 4 (no windows there, so it's created), then launches a client, then goes to another workspace (implicitly destroys workspace 4 since it's empty). Once the client starts up, it's desirable to re-create workspace 4 and place the new window there.

Reply:
I've only renamed this variable. It's now recorded as `pw->name` just below instead of `pw->workspace`, since that's what it contains. That scenario still works as intended.

Actually it looks like this strdup line really is unnecessary though, since I do the same in L141 in this commit, and don't even remember to free the old name str.